python/cudf/cudf/core/multiindex.py

```diff
@@ -1649,3 +1670,46 @@ def get_loc(self, key, method=None, tolerance=None):
         mask = cupy.full(self._data.nrows, False)
         mask[true_inds] = True
         return mask
+
+    def _union(self, other, sort=None):
+        other_df = other.copy(deep=True).to_frame(index=False)
+        self_df = self.copy(deep=True).to_frame(index=False)
+        col_names = list(range(0, self.nlevels))
+        self_df.columns = col_names
+        other_df.columns = col_names
+        self_df["order"] = self_df.index
+        other_df["order"] = other_df.index
+
+        result_df = self_df.merge(other_df, on=col_names, how="outer")
+        result_df = result_df.sort_values(
+            by=result_df.columns[self.nlevels :], ignore_index=True
+        )
+
+        midx = MultiIndex.from_frame(result_df.iloc[:, : self.nlevels])
+        midx.names = self.names if self.names == other.names else None
```

Comment:
Aside from these two lines this function looks almost identical to `BaseIndex._union`. Is it possible to exploit `super()._union` here? If necessary, perhaps by moving a small amount of logic from `BaseIndex._union` into `GenericIndex._union` (the part that isn't applicable to `MultiIndex`)?